src/AppInstallerCLICore/Workflows/GetRequestedUrlFromManifest.cpp

```diff
@@ -0,0 +1,56 @@
+#include "pch.h"
+#include "GetRequestedUrlFromManifest.h"
+
+#include "ManifestComparator.h"
+#include "ShellExecuteInstallerHandler.h"
+#include "WorkflowBase.h"
+
+using namespace AppInstaller::CLI;
+using namespace AppInstaller::Utility;
+using namespace AppInstaller::Manifest;
+
+namespace AppInstaller::CLI::Workflow
+{
+    void GetRequestedUrlFromManifest(Execution::Context& context)
+    {
+        const auto& manifest = context.Get<Execution::Data::Manifest>();
+        Workflow::ManifestComparator manifestComparator(context.Args);
+        const auto selectedLocalization = manifestComparator.GetPreferredLocalization(manifest);
+        if (context.Args.Contains(Execution::Args::Type::Homepage)) {
+            if (selectedLocalization.Homepage.empty())
+            {
+                context.Reporter.Info() << Resource::String::UrlNotFound << std::endl;
+            }
+            else {
+                if (!selectedLocalization.Homepage.empty()) {
+                    if (Utility::IsUrlRemote(selectedLocalization.Homepage)) {
+                        context.Reporter.Info() << Resource::String::ShowHomepageOpeningUrl << ": " << selectedLocalization.Homepage << std::endl;
+                        context.Add<Execution::Data::ShowUrls>({ selectedLocalization.Homepage.c_str() });
+                    }
+                    else
+                    {
+                        context.Reporter.Info() << Resource::String::ShowHomepageDisplayUrl << ": " << selectedLocalization.Homepage << std::endl;
+                    }
+                }
+            }
+        }
+        if (context.Args.Contains(Execution::Args::Type::License)) {
+            if (selectedLocalization.LicenseUrl.empty())
+            {
+                context.Reporter.Info() << Resource::String::UrlNotFound << std::endl;
+            }
+            else {
+                if (!selectedLocalization.LicenseUrl.empty()) {
+                    if (Utility::IsUrlRemote(selectedLocalization.LicenseUrl)) {
+                        context.Reporter.Info() << Resource::String::ShowLicenseOpeningUrl << ": " << selectedLocalization.LicenseUrl << std::endl;
+                        context.Add<Execution::Data::ShowUrls>({ selectedLocalization.LicenseUrl.c_str() });
```

Comment:
>context.Add<Execution::Data::ShowUrls>({ selectedLocalization.LicenseUrl.c_str() }); [](start = 24, length = 84)

This second time will overwrite the first vector.  Would probably be easiest to always insert an empty vector at the beginning, then always add here:

At top:
context.Add<Execution::Data::ShowUrls>({});

When adding:
context.Get<Execution::Data::ShowUrls>().emplace_back(selectedLocalization.LicenseUrl);